test/functional/readpreference.test.js

```diff
@@ -548,4 +550,102 @@ describe('ReadPreference', function() {
       client.close(done);
     });
   });
+
+  context('hedge', function() {
+    before(function() {
+      this.testHedge = callback => {
```

Comment:
This feels not specific to hedged testing, maybe we can refactor it into a helper function? We have a common [withClient](https://github.com/mongodb/node-mongodb-native/blob/master/test/functional/shared.js#L78) helper, and it seems like what we want here is something like `withMonitoredClient`?